### PR TITLE
Make `broadcast_expand` consistent with normal broadcasting

### PR DESCRIPTION
**Pull request by mudit2812**
**Context:**
`broadcast_expand` puts results for all measurements inside a numpy array, which causes warnings, and also raises errors due to ragged dimensions when there are multiple measurements with different dimensions. There was also missing logic for handling shot vectors.

**Description of the Change:**
* Update `broadcast_expand` to process the results to look _exactly_ the same as they would without `broadcast_expand`, which is to stack broadcasted results, and put all measurement results inside a tuple.
* Added logic to handle shot vectors.

**Benefits:**
`broadcast_expand` results are at partiy with results using native broadcasting.

**Possible Drawbacks:**
This change is breaking some tests, which will have to be updated before the change can be merged.

**Related GitHub Issues:**

### CODE REVIEW

**Comment by trbromley** [comment]
@mudit2812 do you have a sentence or two summary of what's happening here?

**Comment by mudit2812** [comment]
> @mudit2812 do you have a sentence or two summary of what's happening here?

@trbromley I updated the PR description with more details of what I did. I will add an example later for more context.

File: pennylane/transforms/broadcast_expand.py [hunk 1]
```diff
@@ -140,14 +140,26 @@ def null_postprocessing(results):
             output_tapes.append(new_tape)
 
         def processing_fn(results: qml.typing.ResultBatch) -> qml.typing.Result:
-            if len(tape.measurements) > 1:
-                processed_results = [
-                    qml.math.squeeze(
-                        qml.math.stack([results[b][i] for b in range(tape.batch_size)])
+            if tape._shots.has_partitioned_shots:
```
Comment (albi3ro) [inline pennylane/transforms/broadcast_expand.py:143]:
```suggestion
            if tape.shots.has_partitioned_shots:
```

**Comment by trbromley** [comment]
> This change is breaking some tests, which will have to be updated before the change can be merged.

Thanks! To be on the same page - we decided to either fix this in FF week or shortly after?

**Comment by mudit2812** [comment]
> Thanks! To be on the same page - we decided to either fix this in FF week or shortly after?

This change will not be merged for 0.33

**Comment by mudit2812** [comment]
Closing in favour of #5473

**Pull request closed without merging**